maxon_epos_driver/src/lib/EposMotor.cpp

```diff
@@ -0,0 +1,280 @@
+/**
+ * @file   EposMotor
+ * @brief  
+ * @author arwtyxouymz
+ * @date   2019-06-03 16:15:22
+ */
+
+#include "maxon_epos_driver/Definitions.h"
+#include "maxon_epos_driver/EposMotor.hpp"
+#include "maxon_epos_driver/utils/Macros.hpp"
+#include "maxon_epos_driver/utils/EposException.hpp"
+#include "maxon_epos_driver/control/EposProfilePositionMode.hpp"
+#include "maxon_epos_driver/control/EposProfileVelocityMode.hpp"
+#include "maxon_epos_driver/control/EposCurrentMode.hpp"
+
+#include "maxon_epos_msgs/MotorState.h"
+
+/**
+ * @brief Constructor
+ */
+EposMotor::EposMotor()
+    : m_position(0), m_velocity(0), m_effort(0), m_current(0)
+{}
+
+/**
+ * @brief Destructor
+ *        change the device state to "disable"
+ */
+EposMotor::~EposMotor()
+{
+    try {
+        VCS_NODE_COMMAND_NO_ARGS(SetDisableState, m_epos_handle);
+    } catch (const EposException &e) {
+        ROS_ERROR_STREAM(e.what());
+    }
+}
+
+
+void EposMotor::init(ros::NodeHandle &root_nh, ros::NodeHandle &motor_nh, const std::string &motor_name)
+{
+    m_motor_name = motor_name;
+    initEposDeviceHandle(motor_nh);
+    initProtocolStackChanges(motor_nh);
+
+    // これいる?
+    VCS_NODE_COMMAND_NO_ARGS(SetDisableState, m_epos_handle);
```

Comment:
53行目にSetEnableStateがあるので不要な気がします。
が、EPOSは電源投入時に「CAN接続エラー（実際にはエラーではない）」を送ってくる可能性があるため、
私が作ったC#のコードでは「GetState() -> ClearFault()」を必ず実行していました。（ClearFaultしないと、SetStateに反応しないはず）
もしClearFaultに相当するNODE_COMMANDがあればそちらを使ってもいいかもしれません

Reply:
When should i exec `ClearFault`?
In my code, every connection has `void*` pointer in [NodeHandle](https://github.com/iwata-lab/maxon_epos_ros/blob/multiple_motors/maxon_epos_driver/include/maxon_epos_driver/Device.hpp#L62) object after `OpenDevice` or `OpenSubDevice`.

I think it's better to exec it every time after each devices opened.
Another way is just one time execution to master handle after all devices opened.
What do you think?

Reply:
>I think it's better to exec it every time after each devices opened.
I agree with your idea. Since ClearFault needs node_id, I think it is good to execute in NodeHandle ctor or CreateEposHandle().

Reply:
Fixed.